### ISSUE TITLE
Hyperparameter tuning

### ISSUE THREAD

**Issue by batman7090** [ultralytics/ultralytics#4088]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello @glenn-jocher & @all,

I am training yolov8 model with custom dataset with two classes, (has class imbalance) the ratio between classes is 1:3. I am getting the results of map@0.5=0.75 and it goes on decreasing as the epochs increases. what might be the reason for this? 
how can I do hyper parameter tunning, which hyper parameter should I be more concentrated?

why is the validation loss increasing??
also in confusion matrix background class is detected more even if the percentage of background class in whole dataset is 4%!

thank you in advance!

![confusion_matrix_normalized](https://github.com/ultralytics/ultralytics/assets/51845526/0389d280-4217-4906-9813-761586ea5f1b)
![WhatsApp Image 2023-08-01 at 12 17 25](https://github.com/ultralytics/ultralytics/assets/51845526/4f534346-1c6b-4d75-9777-ce2b8eb96487)

**Comment by glenn-jocher** [comment]
@batman7090 hello,

Thank you for your detailed question.

Firstly, decreasing mAP across epochs could potentially be due to overfitting. You might want to investigate if your model is learning too well to recognize the training data, but failing to generalize it to the validation data.

When it comes to hyperparameter tuning, you'll want to pay attention to a few key parameters. Learning rate is always a crucial one to consider, but don't neglect others like weight decay and batch size. These can often have surprisingly large impacts on your results.

Increasing validation loss over the epochs is an indication that the model is overfitting. The model learns the training data and becomes better at predicting it, thus reducing training loss. However, since it's not generalizing well to unseen data, the validation loss increases.

As for class imbalance, it's common for models to have trouble when the classes in your dataset are not represented equally. The model could be biased towards the class that has more examples. There are a few techniques to deal with this like oversampling the minority class, undersampling the majority class, or using class weights in your loss function to make sure that errors on less common classes are given more importance.

The background class being detected more could be due to the model not being able to distinguish between the classes effectively. This could be improved by either obtaining more data, augmenting the existing data, or tuning the model parameters.

Hyper-parameter tuning, understanding class imbalances, and overcoming overfitting are broad and complex topics in machine learning. Ideally, you will want to experiment with these techniques to see what works best for your dataset and model.

Best of luck with your training!
